core/scheduler/scheduler.go

```diff
@@ -374,6 +380,66 @@ func (s *Scheduler) resolveProDuties(ctx context.Context, slot core.Slot, vals v
 	return nil
 }
 
+// resolveSyncCommDuties resolves sync committee duties for the validators in the given slot's epoch, caching the results.
+func (s *Scheduler) resolveSyncCommDuties(ctx context.Context, slot core.Slot, vals validators) error {
+	duties, err := s.eth2Cl.SyncCommitteeDuties(ctx, eth2p0.Epoch(slot.Epoch()), vals.Indexes())
+	if err != nil {
+		return err
+	}
+
+	var subscriptions []*v1.SyncCommitteeSubscription
+
+	for _, syncCommDuty := range duties {
+		var (
+			startSlot = slot
+			endSlot   = slot.LastSlotInEpoch()
+			vIdx      = syncCommDuty.ValidatorIndex
+		)
+
+		pubkey, ok := vals.PubKeyFromIndex(vIdx)
+		if !ok {
+			log.Warn(ctx, "Ignoring unexpected sync committee duty", nil, z.U64("vidx", uint64(syncCommDuty.ValidatorIndex)), z.I64("slot", slot.Slot))
+			continue
+		}
+
+		// When assigned to an epoch sync committee, signatures must be produced and broadcast for slots on
+		// range [compute_start_slot_at_epoch(epoch) - 1, compute_start_slot_at_epoch(epoch) + SLOTS_PER_EPOCH - 1) rather than for the
+		// range [compute_start_slot_at_epoch(epoch), compute_start_slot_at_epoch(epoch) + SLOTS_PER_EPOCH).
+		// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/validator.md#sync-committee
+		if slot.FirstInEpoch() {
+			startSlot = slot.Previous()
+		}
+
+		for sl := startSlot; sl.Slot <= endSlot.Slot; sl = sl.Next() {
+			// Schedule sync committee contribution aggregation.
+			duty := core.NewSyncContributionDuty(sl.Slot)
+			s.setDutyDefinition(duty, pubkey, core.NewSyncCommitteeDefinition(syncCommDuty))
+		}
+
+		log.Info(ctx, "Resolved sync committee duty",
+			z.U64("vidx", uint64(syncCommDuty.ValidatorIndex)),
+			z.Any("pubkey", pubkey),
+			z.U64("epoch", uint64(slot.Epoch())),
+		)
+
+		subscriptions = append(subscriptions, &v1.SyncCommitteeSubscription{
+			ValidatorIndex:       vIdx,
+			SyncCommitteeIndices: syncCommDuty.ValidatorSyncCommitteeIndices,
+			UntilEpoch:           eth2p0.Epoch(slot.NextEpoch()), // Renew subscription again next epoch.
+		})
+	}
+
+	if len(subscriptions) > 0 {
+		// Subscribe to sync committee subnets.
+		err = s.eth2Cl.SubmitSyncCommitteeSubscriptions(ctx, subscriptions)
```

Comment:
`fetcher` needs this since it queries BN to get sync comm contribution (`/eth/v1/validator/sync_committee_contribution`)